pkg/network/status/status.go

```diff
@@ -186,68 +177,59 @@ func (m *StatusProber) IsReady(ia *v1alpha1.Ingress, gw map[v1alpha1.IngressVisi
 	}
 
 	var workItems []*workItem
-	for gatewayName, hosts := range ingress.HostsPerVisibility(ia, gw) {
-		gateway, err := m.getGateway(gatewayName)
-		if err != nil {
-			return false, fmt.Errorf("failed to get Gateway %q: %w", gatewayName, err)
-		}
-		targetsPerPod, err := m.listGatewayTargetsPerPods(gateway)
-		if err != nil {
-			return false, fmt.Errorf("failed to list the probing URLs of Gateway %q: %w", gatewayName, err)
-		}
-		if len(targetsPerPod) == 0 {
-			continue
+
+	allProbeTargets, err := m.targetLister.ListProbeTargets(ctx, ia)
+	if err != nil {
+		return false, err
+	}
+
+	for ip, targets := range allProbeTargets {
+		// Each Pod is probed using the different hosts, protocol and ports until
+		// one of the probing calls succeeds. Then, the Pod is considered ready and all pending work items
+		// scheduled for that pod are cancelled.
+		ctx, cancel := context.WithCancel(ingCtx)
+		podState := &podState{
+			successCount: 0,
+			context:      ctx,
+			cancel:       cancel,
 		}
 
-		for ip, targets := range targetsPerPod {
-			// Each Pod backing a Gateway is probed using the different hosts, protocol and ports until
-			// one of the probing calls succeeds. Then, the Pod is considered ready and all pending work items
-			// scheduled for that pod are cancelled.
-			ctx, cancel := context.WithCancel(ingCtx)
-			podState := &podState{
-				successCount: 0,
-				context:      ctx,
-				cancel:       cancel,
-			}
+		// Save the podState to be able to cancel it in case of Pod deletion
+		func() {
+			m.mu.Lock()
+			defer m.mu.Unlock()
+			m.podStates[ip] = podState
```

Comment:
The issue you were talking about was that either:
 - We might have two envoys in the same pod (same ip, different ports) that receive programming at different times, or
 - The envoy at a given ip/port combo might not receive all of the host programming atomically

Is that right?  Can you open an issue for this and `// TODO(#1234)` it?

Reply:
Yes, added.